contracts/modules/STO/USDTiered/USDTieredSTO.sol

```diff
@@ -278,20 +297,28 @@ contract USDTieredSTO is USDTieredSTOStorage, STO {
         uint256 tempSold;
         uint256 remainingTokens;
         for (uint256 i = 0; i < tiers.length; i++) {
-            remainingTokens = tiers[i].tokenTotal.sub(tiers[i].mintedTotal);
+            remainingTokens = tiers[i].tokenTotal.sub(tiers[i].totalTokensSoldInTier);
             tempReturned = tempReturned.add(remainingTokens);
-            tempSold = tempSold.add(tiers[i].mintedTotal);
+            tempSold = tempSold.add(tiers[i].totalTokensSoldInTier);
             if (remainingTokens > 0) {
-                tiers[i].mintedTotal = tiers[i].tokenTotal;
+                tiers[i].totalTokensSoldInTier = tiers[i].tokenTotal;
             }
         }
-        address walletAddress = (treasuryWallet == address(0) ? IDataStore(getDataStore()).getAddress(TREASURY) : treasuryWallet);
+        address walletAddress = getTreasuryWallet();
         require(walletAddress != address(0), "Invalid address");
+        if (preMintAllowed)
+            tempReturned = securityToken.balanceOf(address(this));
         uint256 granularity = securityToken.granularity();
         tempReturned = tempReturned.div(granularity);
         tempReturned = tempReturned.mul(granularity);
-        securityToken.issue(walletAddress, tempReturned, "");
-        emit ReserveTokenMint(msg.sender, walletAddress, tempReturned, currentTier);
+        if (tempReturned != uint256(0)) {
+            if (preMintAllowed) {
+                securityToken.transfer(walletAddress, tempReturned);
```

Comment:
Maybe we should have an event emitted here as well - i.e. `ReserveTokenTransfer` - this is already kind of captured with the token transfer itself, but maybe good to be explicit that it has been finalized.

Reply:
Fixed